app/controllers/UserController.scala

```diff
@@ -195,13 +200,45 @@ class UserController @Inject()(userService: UserService,
         Fox.successful(())
     })
 
-  private def checkAdminOnlyUpdates(user: User,
-                                    isActive: Boolean,
-                                    isAdmin: Boolean,
-                                    isDatasetManager: Boolean,
-                                    oldEmail: String,
-                                    email: String)(issuingUser: User): Boolean =
-    if (isActive && user.isAdmin == isAdmin && oldEmail == email && isDatasetManager == user.isDatasetManager)
+  private def checkTeamManagerOnlyUpdates(user: User,
+                                          experiences: Map[String, Int],
+                                          oldExperiences: Map[String, Int],
+                                          teams: List[TeamMembership],
+                                          oldTeams: List[TeamMembership])(issuingUser: User): Fox[Boolean] =
+    if (experiences == oldExperiences && teams == oldTeams)
+      Fox.successful(true)
+    else userService.isEditableBy(user, issuingUser)
+
+  private def checkPasswordIfEmailChanged(user: User, passwordOpt: Option[String], oldEmail: String, email: String)(
+      issuingUser: User)(implicit m: MessagesProvider): Fox[Unit] =
+    if (oldEmail == email) {
+      Fox.successful(())
+    } else if (user._id == issuingUser._id) {
+      passwordOpt match {
+        case Some(password) =>
+          val credentials = Credentials(user._id.id, password)
+          Fox.fromFutureBox(
+            credentialsProvider
+              .authenticate(credentials)
+              .flatMap { loginInfo =>
+                userService.retrieve(loginInfo).map {
+                  case Some(user) => Full(())
+                  case None       => Failure(Messages("error.noUser"))
+                }
+              }
+              .recover {
+                case _: ProviderException =>
+                  Failure(Messages("error.invalidCredentials"))
+              })
+        case None => Fox.failure(Messages("error.passwordsDontMatch"))
```

Comment:
The error message doesn’t quite fit, maybe introduce a new one.
Also you could try what happens if you skip the Messages invocation and return the string directly. If I remember correctly, all strings in failures will be automatically passed through Messages by our Controllers.